InstagramStories/Source/CollectionViewCell/IGStoryPreviewCell.swift

```diff
@@ -450,7 +456,29 @@ final class IGStoryPreviewCell: UICollectionViewCell, UIScrollViewDelegate {
             }
         }
     }
-    
+    func getProgressView(with index: Int) -> IGSnapProgressView? {
+        let progressView = storyHeaderView.getProgressView
+        if progressView.subviews.count > 0 {
+            let pv = getProgressIndicatorView(with: index)?.subviews.first as? IGSnapProgressView
+            guard let currentStory = self.story else {
+                fatalError("story not found")
+            }
+            pv?.story = currentStory
+            return pv
+        }
+        return nil
+    }
+    func getProgressIndicatorView(with index: Int) -> UIView? {
+        let progressView = storyHeaderView.getProgressView
+        if progressView.subviews.count>0 {
```

Comment:
Can you try Nil Colacing operator like return Array.filter{}.first ?? nil

Reply:
done